app/forms/keywords_form.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsForm
+  include ActiveModel::Model
+
+  validates_with KeywordsFormValidator
+
+  attr_reader :file, :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(csv_file)
+    @file = csv_file
+
+    return false if invalid?
+
+    begin
+      save_keywords_to_db
+    rescue ActiveRecord::ActiveRecordError
+      errors.add('Invalid File')
+    end
+
+    errors.empty?
+  end
+
+  private
+
+  def save_keywords_to_db
+    keyword_records = parse_keywords.map { |keyword| add_keyword_record(keyword) }
+
+    SaveKeywordsToDb.new(keyword_records).call
+  end
+
+  def parse_keywords
+    csv_data = CSV.read(file)
+    csv_data.map(&:first)
```

Comment:
This can be

```suggestion
  def parse_keywords
    csv_data = CSV.read(file)
    keywords = csv_data.map(&:first)
```

Reply:
Just a question, why would we do this though? 🤔 I was under the assumption that it wouldn't need to be assigned to a variable right? 😮

Reply:
It's related to [this suggestion](https://github.com/aetheryna/google-scraper-ruby/pull/34#discussion_r997908113) to use `attr_accessor`, we can declare the `keywords` as a form attribute then assign the value here, after that we can use it in other method

```ruby
attr_accessor :keywords

def parse_keywords
  keywords = ... // parse the keywords from the CSV file then assign the value here
end

def save_keywords
  // Then we can use the attribute here
end

```

Reply:
Updated in https://github.com/aetheryna/google-scraper-ruby/pull/34/commits/439908c2734f5d8ffecd8a64310ecddee29f9cf6 👷🏾